package.json

```diff
@@ -39,6 +39,11 @@
       "import": "./optics.module.js",
       "require": "./optics.js",
       "types": "./optics.d.ts"
+    },
+    "./xstate": {
+      "import": "./xstate.module.js",
+      "require": "./xstate.js",
```

Comment:
XState has some module state and thus providing both import & require conditions can lead to problems due to the so-called "dual package hazard" if one mixes ESM & CJS in their app. For a package like XState this is somewhat minimized as it's primarily consumed by applications and their code usually sticks to a single module format. Nevertheless, it's worth knowing that this exists and that this is not 100% safe to do here.

Given that problem you should consider adding a `"module"` condition - it would precede other ones and just point to `./xstate.module.js`. This would help bundlers, like webpack 5, to always use a single copy of `jotai/xstate` - regardless of the format of the file that loads it.

Reply:
I have heard of "dual package hazard".

This is somewhat experimental and I wanted to discuss with someone. It seems like the bundlers like webpack doesn't read `"exports"` at the moment?

Not sure if `"module"` condition is allowed? At least not in https://nodejs.org/api/packages.html#packages_conditional_exports .
I don't think our approach of `jotai/*` bundles is industry standard and not sure if we can make it totally safe. But, we would like to make it safer as much as possible.

Reply:
> This is somewhat experimental and I wanted to discuss with someone. It seems like the bundlers like webpack doesn't read "exports" at the moment?

webpack@5 and @rollup/plugin-node-resolve@11 read that.

> Not sure if "module" condition is allowed? At least not in https://nodejs.org/api/packages.html#packages_conditional_exports .

This is a condition "defined" outside of node's documentation and was coined by Tobias from webpack (as a result of discussing the issue with other people interested in the topic) - https://webpack.js.org/guides/package-exports/#conditions

Reply:
Ohhhh! I didn't follow the latest feature from webpack.

> https://webpack.js.org/guides/package-exports/#conditions

OK, got it. I think we should specify "module" and "require" as I'm not really sure if it really supports all "import" cases.

Thanks soooo much for pointing this out.